### PR TITLE
feat: track feature missingness rates

### PR DESCRIPTION
**Pull request by ireneisdoomed**
This PR includes:
- new `L2GFeatureMatrix.calculate_feature_missingness_rate` to calculate the rate at which the annotation of a feature in the training set is missing, as suggested by @addramir. This is a nice quality control metric, and we could even use it to correlate it with the model performance.
- addition of `L2GFeatureMatrix.features_list` attribute.
- minor change: removal of the evaluation logic outside W&B. This showed an incomplete picture, and it was only useful for testing. It makes more sense to fully track the experiment and then delete it in the dashboard.

image
Link to dashboard: https://wandb.ai/open-targets/otg_l2g?workspace=user-opentargets

### CODE REVIEW

**Comment by ireneisdoomed** [comment]
@addramir I didn't track number of missing values for simplicity. It can be derived, as we are also tracking the training data.

File: src/otg/method/l2g/model.py [hunk 1]
```diff
@@ -218,30 +223,7 @@ def evaluate(
             labelCol="label", predictionCol="prediction"
         )
 
-        print("Evaluating model...")  # noqa: T201
```
Comment (d0choa) [inline src/otg/method/l2g/model.py:221]: nice cleanup

File: src/otg/dataset/l2g_prediction.py [hunk 2]
```diff
@@ -71,8 +74,9 @@ def from_credible_set(
             _df=(
                 LocusToGeneModel.load_from_disk(
                     model_path,
-                    features_list=fm.df.drop("studyLocusId", "geneId").columns,
```
Comment (d0choa) [inline src/otg/dataset/l2g_prediction.py:74]: much cleaner

**Review by d0choa — APPROVED** [review]
